jetstream/bigquery_client.py

```diff
@@ -108,13 +113,32 @@ def execute(
 
         if destination_table:
             # add a label with the current timestamp to the table
-            self.add_labels_to_table(
-                destination_table,
-                {"last_updated": self._current_timestamp_label()},
-            )
+            labels = {"last_updated": self._current_timestamp_label()}
+            if experiment_slug:
+                labels["experiment_slug"] = experiment_slug
+            self.add_labels_to_table(destination_table, labels)
 
         return job
 
+    def tables_matching_label(self, label_value: str, label_key: str = "experiment_slug"):
+        """Returns a list of tables matching the specified label."""
+        job = self.client.query(
+            rf"""
+            SELECT
+                table_name
+            FROM
+                {self.dataset}.INFORMATION_SCHEMA.TABLE_OPTIONS
+            WHERE
+                option_name = 'labels'
+                AND COALESCE(REGEXP_EXTRACT(
+                    option_value,
+                    '.*STRUCT\\(\"{label_key}\", \"([^\"]+)\"\\).*'
+                ), '') = '{label_value}'
+            """
+        )
+        result = list(job.result())
+        return [row.table_name for row in result]
```

Comment:
idk if we want to leave a comment here about why we are running this slightly cryptic query instead of using the Python SDK?

-> doing something like `client.get_table(table).labels.get` is a lot slower when done a couple hundred times (although it would be a lot more readable.)

Reply:
Good idea, I'll add a comment.

And yea exactly, I didn't see a way to get all tables by label with the SDK, and it seemed inefficient to get every table individually to check its labels.